# Overview of  Processing Stages

###Multiple Genome Alignment

The explanation of strategy and primary procedures is presented below.  

**Repeat mask：** All of the sequences used to construct alignments should be identified and annotated against a repeats library in order to make all repetitive elements masked. This procedure is operated to ensure that align programs will not be confused by similar sequences. The software we use here is RepeatMasker, and it may take a relative long run time for large sequences.

**Partition sequences:** Chromosomal sequences will be divided into several small files if their length exceeds the maximum length of 500000bp. This step is necessary since comparison between two over-length sequences will seriously reduce the processing efficiency. 

**Pairwise alignment:** Sequences from different species are firstly pairwise aligned with Blastz or its improved version Lastz. It’s previous version was initially designed as underlying alignment program of PipMaker server and has been widely employed in numerous science websites such as EnsemblGenomes (https://www.ensembl.org/) and UCSC Genome Browser (https://genome.ucsc.edu/). A distinct advantage of Blastz/Lastz is its sensitivity in alignment for a large percentage of neutrally evolving regions. Since many genomic sequences have no apparent traces of natural selection, most existing programs are not competent to detect those segments at genome-wide scale. This step takes FASTA sequences as input and produces output in its own format lav.

**Chaining and netting:** After primary alignments, we apply the chaining algorithms to building chains of related fragments. Since many fragments might match with several others, the strategy is to keep the longest one. Chaining is also used to eliminate some redundant alignments and poor quality fragments. Then, extended chain alignments are further linked into larger blocks by netting. The format of the result files are axt. 

**Mafing:** Convert previous step result to MAF format. MAF format contains genomic location information for aligning sequence blocks, which is convenient for further analysis.

**Construct phylogenetic tree:** Users may choose to submit a phylogenetic tree of input genomes, or to make use of the phylogenetic tree constructed by our programs for next step. The tree will provide a reference for the combination of multiple alignments.

**Combine multiple pairwise results:** Pairwise alignments are grouped to generate whole genome multiple alignments by Multiz. Multiz is a phylogenetic tree directed alignment tool, it uses the tree constructed previously and start with sequences closely related on evolution scale. This step is not needed if we just have two species.

**Re-aligning:** Since the combination of pairwise alignments may give rise to new gaps and alignment disruption, we take another process of re-align to fix this problem and make sure that sequences of all species are well-aligned. There are three sequence alignment methods for users to select depending on their needs.
  
  
###Self-alignment for paralog detection

Another major function provided in EGA is paralog sequence detection.  

**Self-alignment:** First, genome sequences are aligned to itself to find some similar segments. The aligment tool is as same as multiple alignment described above. 

**Chaining:** related fragments are joined together by chaining algorithms to build longer candidates.  

**Two-wheeled blast:** n order to explore similar sequences at the most extent, candidates being chained above are used to perform two-wheeled blast against all of the whole genome.  

**Filter and cluster:** Pairs of similar sequences are then being filtered and clustered by coverage and similarity. Pairs which make up of more than 90% of each other and with a similarity value greater than 90% will be finally preserved.  