# Part A: Online Data Analysis Platform

Online data analysis platform is constructed for pairwise/multiple alignments with genomes smaller than 20M. We provide user-friendly interfaces and detailed instructions. Users can visit online EGA through......

This image shows a screen snapshot of our website Home Page. The name of our application is Easy Genome Aligner. Menu bars on the top  show the main buttons, going from left to right as: Home, Upload, Align, Process, Contact. At the top right-hand corner shows two links for you to log in or register a new ID. 

You can either register a new  ID or log into the public ID with the username "guest" and password "password".

Two major functions of our application are multiple genome alignment and paralog detection via self-alignment. You could click on the “view details” button for more information. Following are three example descriptions corresponding to different purposes of users.

####**Example 1: Yeast Genomes Alignment**

Let’s take this unicellular eukaryotic model organism for example and provide a detailed description. 

We are going to compare nine yeast strains, which are S288c, Lachancea kluyveri, Saccharomyces bayanus, Saccharomyces bayanus var. uvarum, Saccharomyces castellii, Saccharomyces kudriavzevii, Saccharomyces mikatae, Saccharomyces paradoxus, Saccharomyces pastorianus respectively. Sequences can be found in the download section of http://www.yeastgenome.org/download-data/sequence.

![alt text](/Users/liuliwei/Desktop/EGA submission/pic/yeast_genome.png "yeast_genome")

**Data download and file format change**
 Rename the downloaded files with a brief name, which will also be used as the genome name in result files. We provide a command line for file format processing. Files processed can be used for alignment.

>perl -ne "if(/^\>/){s/\W+/_/g; s/^\_/>/; s/\_$/\n/;} print;" example.fasta >out.fa
	
Notice that this command can process most of the sequences that are not downloaded form NCBI. For those sequences downloaded from NCBI, our website can process them automatically. 

For users' convenience, we have provided a test account stored with the data used for example tests. You can log into the account for yeast with the username "bacteria" and password "password".

**Upload**
click on "Upload" to enter the page after you log in the site. The upload page shows an upload dialog. If you need to upload a genome file, you can select a file in the open dialog box, and click on the “Upload” button. When you are doing a web-based alignment instead of in the VM, the individual file size is limited to 30M. Only three kinds of file types are accepted: .fasta, .fasta.gz, .newick. You can click on the red button to remove an uploaded file. _**Do not**_ refresh the page while uploading a file!

**Alignment**
You can click on the “Align” button on the menu to enter the align page. At this page you should set some parameters for the construction of alignments. Some parameters are required, such as the name of this alignment job, a target genome, at least one query genome. We have provided the recommended values for Minimum fragment length and a realigning method. Besides, you can choose to provide a guide tree or not. The absence of a guide tree will take a while for us to generate one. 

After parameter settings are completed, you can click on “Begin aligning” button to jump to the processing page automatically.

This page displays three dialog boxes. The dialog above contains parameters set in the previous step. You can’t open a new job if the current one has not finished. However, when your job is completed or you want this job terminated, you can click on the red button at the top-right corner and open up a new task. 
The results are provided for you to download but terminated job can’t recover.

The middle dialog contains buttons for different processing stages. Among them, the first step is data pre-processing. You can click on the “prepare.sh” to start a job running. The information section contains general information about the running task including the start and end times and running status. 

Below is progress information for the particular run step. When the last line shows “\***DONE***”, the step is supposed to be completed. Users may refresh this page and move on to the next step. 

After the pre-processing completes, refresh the page and buttons for the rest steps will appear in the middle dialog box. Users may click on buttons in sequence  to generate an alignment. For every step, when the last line shows “\***DONE***”, the task is completed and you can move on to the next step.

**Downloading results**
After all jobs are done, users may click on the red “finish” button and download the results generated in every step.（对于这9个酵母基因组比对的用时情况如下所示。  

####**Example 2: Bacteria Genomes Alignment**

**Data download**  
There are two perl programs  provided to help users to download genome sequences from NCBI conveniently. The basic syntax are as shown below.  

*Get_seq.pl* is used for downloading individual genome sequence.  

>perl ~/Scripts/withncbi/util/batch_get_seq.pl \$genome_id $dir  
	
*Batch_get_seq.pl* is used for downloading a batch of genome sequences at a time.  

>perl ~/Scripts/withncbi/util/batch_get_seq.pl -r -p -f yeast_name_seq.csv 2>&1 | tee yeast_name_seq.log

Genome sequences downloaded from NCBI need not the step to change the formats of headers. Downloaded genomes are named as the corresponding RefSeq numbers in NCBI, which will also be processed as the genome name in result files.

We also provided a test account stored with the data used for example tests. You can log into the account for bacteria with the username "bacteria" and password "password".

The process of alignment is similar to that of yeast. You can skip this part if you have read the instruction for yeast. If not, the instruction below will help you get a hands on start.

**Upload**
click on "Upload" to enter the page after you log in the site. The upload page shows an upload dialog. If you need to upload a genome file, you can select a file in the open dialog box, and click on the “Upload” button. When you are doing a web-based alignment instead of in the VM, the individual file size is limited to 30M. Only three kinds of file types are accepted: .fasta, .fasta.gz, .newick. You can click on the red button to remove an uploaded file. **_Do not_** refresh the page while uploading a file!

**Alignment**
You can click on the “Align” button on the menu to enter the align page. At this page you should set some parameters for the construction of alignments. Some parameters are required, such as the name of this alignment job, a target genome, at least one query genome. We have provided the recommended values for Minimum fragment length and a realigning method. Besides, you can choose to provide a guide tree or not. The absence of a guide tree will take a while for us to generate one. 

After parameter settings are completed, you can click on “Begin aligning” button to jump to the processing page automatically.

This page displays three dialog boxes. The dialog above contains parameters set in the previous step. You can’t open a new job if the current one has not finished. However, when your job is completed or you want this job terminated, you can click on the red button at the top-right corner and open up a new task. 
The results are provided for you to download but terminated job can’t recover.

The middle dialog contains buttons for different processing stages. Among them, the first step is data pre-processing. You can click on the “prepare.sh” to start a job running. The information section contains general information about the running task including the start and end times and running status. 

Below is progress information for the particular run step. When the last line shows “\***DONE***”, the step is supposed to be completed. Users may refresh this page and move on to the next step. 

After the pre-processing completes, refresh the page and buttons for the rest steps will appear in the middle dialog box. Users may click on buttons in sequence  to generate an alignment. For every step, when the last line shows “\***DONE***”, the task is completed and you can move on to the next step.

**Downloading results**
After all jobs are done, users may click on the red “finish” button and download the results generated in every step.

####**Example 3: Self-alignment for Paralog Detection**

**Data download and file format change**
Here we take yeast S288c as an example. Sequences can be found in the download section of http://www.yeastgenome.org/download-data/sequence.

Rename the downloaded file with a brief name. This name will also be used as the genome name in result files. We provide a command line for file format processing. Files processed can be used for alignment.

>perl -ne "if(/^\>/){s/\W+/_/g; s/^\_/>/; s/\_$/\n/;} print;" example.fasta

We also provided a test account stored with the data used for example tests. You can log into the account for paralog alignment with the username "paralog" and password "password".

**Upload**
click on "Upload" to enter the page after you log in the site. The upload page shows an upload dialog. If you need to upload a genome file, you can select a file in the open dialog box, and click on the “Upload” button. When you are doing a web-based alignment instead of in the VM, the individual file size is limited to 30M. Only three kinds of file types are accepted: .fasta, .fasta.gz, .newick. You can click on the red button to remove an uploaded file. _**Do not**_ refresh the page while uploading a file!

**Alignment**
You can click on the “Align” button on the menu to enter the align page. At this page you should set some parameters for the construction of alignments. Some parameters are required, such as the name of this alignment job and a target genome. In order to do paralog detection, you should check the “self alignment” box so that you cannot choose query genomes. We have provided the recommended values for Minimum fragment length and a realigning method. Besides, you can choose to provide a guide tree or not. The absence of a guide tree will take a while for us to generate one. 

After parameter settings are completed, you can click on “Begin aligning” button to jump to the processing page automatically.

This page displays three dialog boxes. The dialog above contains parameters set in the previous step. You can’t open a new job if the current one has not finished. However, when your job is completed or you want this job terminated, you can click on the red button at the top-right corner and open up a new task. 
The results are provided for you to download but terminated job can’t recover.

The middle dialog contains buttons for different processing stages. Among them, the first step is data pre-processing. You can click on the “prepare.sh” to start a job running. The information section contains general information about the running task including the start and end times and running status. 

Below is progress information for the particular run step. When the last line shows “\***DONE***”, the step is supposed to be completed. Users may refresh this page and move on to the next step. 

After the pre-processing completes, refresh the page and buttons for the rest steps will appear in the middle dialog box. Users may click on buttons in sequence  to generate an alignment. For every step, when the last line shows “\***DONE***”, the task is completed and you can move on to the next step.

**Downloading results**
After all jobs are done, users may click on the red “finish” button and download the results generated in every step.